subsys/bluetooth/host/ecc.c

```diff
@@ -166,6 +166,10 @@ int bt_dh_key_gen(const uint8_t remote_pk[64], bt_dh_key_cb_t cb)
 		return -EADDRNOTAVAIL;
 	}
 
+	if (!memcmp(remote_pk, pub_key, 64)) {
+		return -EINVAL;
+	}
```

Comment:
SMP already does this in smp.c, perhaps mesh should do it itself as well?

Reply:
Moved to mesh